testing/citest/tests/openstack_smoke_test.py

```diff
@@ -0,0 +1,265 @@
+    OpenStack client to work. Please refer testing/citest/README.md for more
+    details.
+
+Sample Usage:
+    PYTHONPATH=testing/citest \
+    python tesing/citest/tests/openstack_smoke_test.py \
+    --native_hostname=host-running-smoke-test
+"""
+
+# Standard python modules.
+import sys
+
+# citest modules.
+import citest.openstack_testing as os
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+import citest.base
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+class OpenStackSmokeTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the smoke test.
+
+  This scenario defines the different test operations.
+  We're going to:
+    Create a Spinnaker Application
+    Create a Spinnaker Security Group
+    Delete each of the above (in reverse order)
+  """
+
+  @classmethod
+  def new_agent(cls, bindings):
+    """Implements citest.service_testing.AgentTestScenario.new_agent."""
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 180
+    return agent
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    """
+    super(OpenStackSmokeTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    # We'll call out the app name because it is widely used
+    # because it scopes the context of our activities.
+    # pylint: disable=invalid-name
+    self.TEST_APP = bindings['TEST_APP']
+    self.TEST_STACK = bindings['TEST_STACK']
+    self.TEST_SECURITY_GROUP = 'sec_grp_'+ bindings['TEST_APP']
+
+
+  def create_app(self):
+    """Creates OperationContract that creates a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_create_app_operation(
+            bindings=self.bindings, application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_OS_ACCOUNT']),
+        contract=contract)
+
+  def delete_app(self):
+    """Creates OperationContract that deletes a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_delete_app_operation(
+            application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_OS_ACCOUNT']),
+        contract=contract)
+
+  def create_a_security_group(self):
+    """Creates OsContract for createServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was created.
+    """
+    rules = [{
+        "fromPort":80,
+        "toPort":80,
+        "prevcidr":"0.0.0.0/0",
+        "cidr":"0.0.0.0/0",
+        "ruleType":"TCP",
+        "remoteSecurityGroupId":"",
+        "icmpType":-1,
+        "icmpCode":-1
+        },
+             {
```

Comment:
indentation is weird. These should be same level as first. Also space after the ':' in the key.